### PR TITLE
WIP: Feat/Stripe: Added more object of the stripe products categorie

### PR DESCRIPTION
**Pull request by nielsbrakel**
NOTE: WIP Just looking for some feedback :)

## What kind of change does this PR introduce?

1. Added new product objects: coupons, promotion_codes, tax_codes, tax_rates and shipping rates

## What is the current behavior?

Added more object to make the wrapper more complete

## What is the new behavior?

You can now access more information from stripe

## Additional context

This is still a WIP so will place some comments on my own code to check some things.

### CODE REVIEW

File: wrappers/src/fdw/stripe_fdw/stripe_fdw.rs [hunk 1]
```diff
@@ -97,6 +97,8 @@ fn body_to_rows(
                     .and_then(|v| match *col_type {
                         "bool" => v.as_bool().map(Cell::Bool),
                         "i64" => v.as_i64().map(Cell::I64),
+                        "f64" => v.as_f64().map(Cell::F64),
+                        "json" => v.as_i64().map(Cell::Json),
```
Comment (nielsbrakel) [inline wrappers/src/fdw/stripe_fdw/stripe_fdw.rs:101]: Not sure if this is correct... maybe I should first learn how rust works...

File: wrappers/src/fdw/stripe_fdw/stripe_fdw.rs [hunk 2]
```diff
@@ -156,6 +161,8 @@ fn row_to_body(row: &Row) -> JsonValue {
                         if let Some(m) = v.0.clone().as_object_mut() {
                             map.append(m)
                         }
+                    } else {
+                        map.insert(col_name, v);
```
Comment (nielsbrakel) [inline wrappers/src/fdw/stripe_fdw/stripe_fdw.rs:165]: Curious if someone can shed some light on this code if this is the correct implementation for normal `json` values

File: wrappers/src/fdw/stripe_fdw/stripe_fdw.rs [hunk 3]
```diff
@@ -493,46 +532,106 @@ impl StripeFdw {
...
+                resp_body,
+                vec![
+                    ("id", "string"),
+                    ("active", "bool"),
+                    ("country", "string"),
+                    ("description", "string"),
+                    ("display_name", "string"),
+                    ("inclusive", "bool"),
+                    ("percentage", "f64"),
+                    ("created", "timestamp"),
```
Comment (nielsbrakel) [inline wrappers/src/fdw/stripe_fdw/stripe_fdw.rs:593]: I'm not sure if there is a rule about which values deserve their own column in the wrapper or if they should remain in the attributes column.

File: wrappers/src/fdw/stripe_fdw/stripe_fdw.rs (hunk 1 above)
Comment (burmecia) [reply wrappers/src/fdw/stripe_fdw/stripe_fdw.rs:101]: it is correct for `f64`, but for `json` I think it is better to use string convert in between.

File: wrappers/src/fdw/stripe_fdw/stripe_fdw.rs (hunk 2 above)
Comment (burmecia) [reply wrappers/src/fdw/stripe_fdw/stripe_fdw.rs:165]: It looks correct, but that requires the `col_name` matches with Stripe json field, may need have some testing on this.

File: wrappers/src/fdw/stripe_fdw/stripe_fdw.rs (hunk 3 above)
Comment (burmecia) [reply wrappers/src/fdw/stripe_fdw/stripe_fdw.rs:593]: There is no hard rules, a rule of thumb is more commonly used fields should be extracted to their own columns, and put all the other less common fields in the `attrs` column.

**Comment by burmecia** [comment]
Thanks for the PR! Happy to see more objects to be added to this FDW, let us know once it is ready.

**Comment by psteinroe** [comment]
hey @nielsbrakel, are you still working on this? I would like to add subscription_items and their usage record summaries to the list of supported objects and am wondering if I should start a new pr, or collaborate on this one?

**Comment by nielsbrakel** [comment]
@psteinroe I'm planning to continue working on it, but can not give a date or time.

**Comment by olirice** [comment]
closing due to inactivity. please re-open when ready

**Pull request closed without merging**